### PR TITLE
fix: update ruamel.yaml usage to use supported APIs

### LINKED ISSUE #5515: tests are busted due to something with setuptools and ruamel.yaml
IDK why, but the tests for conda skeleton are now breaking due to the removal of `safe_dump` in `ruamel.yaml >=0.18.0`.

see https://github.com/conda/conda-build/actions/runs/11484656642/job/31962986476

### PR DESCRIPTION
**Pull request by beckermr**
### Description

This PR moves the ruamel.yaml calls to the new APIs.

Fixes #5515 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda-build/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda_build/skeletons/pypi.py [hunk 1]
```diff
@@ -154,7 +154,7 @@
 +    data['classifiers'] = kwargs.get('classifiers', None)
 +    data['version'] = kwargs.get('version', '??PACKAGE-VERSION-UNKNOWN??')
 +    with io.open(os.path.join("{}", "pkginfo.yaml"), 'w', encoding='utf-8') as fn:
-+        fn.write(yaml.safe_dump(data, encoding=None))
++        fn.write(yaml.YAML(typ='safe', pure=True).dump(data, encoding=None))
```
Comment (beckermr) [inline conda_build/skeletons/pypi.py:157]:
```suggestion
+        fn.write(yaml.YAML(typ='safe', pure=True).dump(data))
```

**Comment by beckermr** [comment]
pre-commit.ci autofix

File: tests/test_api_skeleton.py [hunk 2]
```diff
@@ -9,7 +9,6 @@
 from typing import TYPE_CHECKING
 
 import pytest
-import ruamel.yaml
```
Comment (beckermr) [inline tests/test_api_skeleton.py:12]:
```suggestion
import ruamel.yaml
```

File: tests/test_api_skeleton.py [hunk 3]
```diff
@@ -485,7 +484,8 @@ def test_pypi_section_order_preserved(tmp_path: Path):
     ]
 
     # The loader below preserves the order of entries...
-    recipe = ruamel.yaml.load("\n".join(lines), Loader=ruamel.yaml.RoundTripLoader)
+    _yaml = YAML(typ="rt")
```
Comment (beckermr) [inline tests/test_api_skeleton.py:487]:
```suggestion
    _yaml = ruamel.yaml.YAML(typ="rt")
```

File: tests/test_api_skeleton.py [hunk 4]
```diff
@@ -10,7 +10,6 @@
 
 import pytest
 import ruamel.yaml
-
 from conda_build import api
```
Comment (beckermr) [inline tests/test_api_skeleton.py:13]:
```suggestion

from conda_build import api
```

**Comment by beckermr** [comment]
@zooba @conda/builds-tools This PR is ready to go. The test failures are unrelated.

**Review by isuruf — APPROVED** [review]

**Pull request merged**